libcnb/src/layer/public_interface.rs

```diff
@@ -189,6 +193,17 @@ impl<M> LayerResultBuilder<M> {
         self
     }
 
+    #[must_use]
+    pub fn exec_d_program(
```

Comment:
I think this needs a rustdoc that (a) shows example usage (eg with the `additional_buildpack_binary_path!` macro, to encourage people not to try and build their own `PathBuf`), (b) explains what `name` does.

Whilst there is the code in `examples/`, it's not visible when using the IDE.

Plus when I first read the example:

```
        LayerResultBuilder::new(GenericMetadata::default())
            .exec_d_program(
                "dice_roller",
                additional_buildpack_binary_path!("dice_roller"),
            )
            .build()
```

...it left me wondering what the difference was between a `name` of `dice_roller` and the string passed into `additional_buildpack_binary_path!`.

I was able to work out what `name` was used for by inspecting `write_layer()` in `libcnb/src/layer/handling.rs`, however it would be good to cover this in the rustdoc instead.

Reply:
Added in 0163719